dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -21,10 +21,63 @@
 
 #include "precomp.h"
 
+#define NTOS_MODE_USER
+#include <ndk/iofuncs.h>
+#include <ndk/obfuncs.h>
+
 WINE_DEFAULT_DEBUG_CHANNEL(shell);
 
 EXTERN_C BOOL PathIsExeW(LPCWSTR lpszPath);
 
+BOOL GetPhysicalFileSize(LPCWSTR PathBuffer, PULARGE_INTEGER Size)
+{
+    UNICODE_STRING FileName;
+    OBJECT_ATTRIBUTES ObjectAttributes;
+    IO_STATUS_BLOCK IoStatusBlock;
+    HANDLE FileHandle;
+    FILE_STANDARD_INFORMATION FileInfo;
+    NTSTATUS Status;
+    
+    if (!RtlDosPathNameToNtPathName_U(PathBuffer, &FileName, NULL, NULL))
+    {
+        ERR("RtlDosPathNameToNtPathName_U failed\n");
+        return FALSE;
+    }
+
+    InitializeObjectAttributes(&ObjectAttributes,
+                               &FileName,
+                               OBJ_CASE_INSENSITIVE,
+                               NULL,
+                               NULL);
+    Status = NtOpenFile(&FileHandle,
+                        FILE_READ_ATTRIBUTES | SYNCHRONIZE,
+                        &ObjectAttributes,
+                        &IoStatusBlock,
+                        FILE_SHARE_READ,
+                        FILE_SYNCHRONOUS_IO_NONALERT);
+    if (!NT_SUCCESS(Status))
```

Comment:
```suggestion

    RtlFreeHeap(RtlGetProcessHeap(), 0, FileName.Buffer);

    if (!NT_SUCCESS(Status))
```

Reply:
`RtlFreeUnicodeString(&FileName);` would be better.

Reply:
Let's go with RtlFreeUnicodeString(&FileName);